### ISSUE TITLE
Adding support for native Huggingface gradient checkpointing

### ISSUE THREAD

**Issue by xhluca** [texttron/tevatron#50]
It would be nice to have support for Huggingface's gradient checkpointing capabilities.
With models like BERT, you can simply do:
```
model.gradient_checkpointing_enable()
```

[Here's the docs on the subject.](https://huggingface.co/docs/transformers/main_classes/model#transformers.PreTrainedModel.gradient_checkpointing_enable)

I think this might be a good option for models that might not be supported by GradCache (or maybe both can be combined to save even more memory, though I'm not sure if they are compatible).

**Comment by MXueguang** [comment]
Hi @xhluca,
I am not very sure about this. maybe @luyug  can take a look.
